src/app/tests/TestReportScheduler.cpp

```diff
@@ -526,39 +526,69 @@ class TestReportScheduler
         NL_TEST_ASSERT(aSuite, syncScheduler.mTestNextReportTimestamp == node1->GetMaxTimestamp());
 
         // Confirm behavior when a read handler becomes dirty
-        readHandler1->ForceDirtyState();
-        // since the min interval on readHandler1 is 0, it should be reportable now
-        NL_TEST_ASSERT(aSuite, syncScheduler.IsReportableNow(readHandler1));
-        readHandler1->mObserver->OnBecameReportable(readHandler1);
-        // Confirm that the next report emission is scheduled on the min timestamp of readHandler2 as it is the highest
-        NL_TEST_ASSERT(aSuite, syncScheduler.mTestNextReportTimestamp == node2->GetMinTimestamp());
+        readHandler2->ForceDirtyState();
+        NL_TEST_ASSERT(aSuite, !syncScheduler.IsReportableNow(readHandler2));
+        NL_TEST_ASSERT(aSuite, !syncScheduler.IsReportableNow(readHandler1));
         // Simulate wait enough for min timestamp of readHandler2 to be reached (1s)
         // Increment the mock system time to the max to the time waited
         sTestTimerSynchronizedDelegate.IncrementMockTimestamp(System::Clock::Milliseconds64(1000));
 
-        // Confirm that readHandler2 is now reportable
         NL_TEST_ASSERT(aSuite, syncScheduler.IsReportableNow(readHandler2));
         readHandler2->mObserver->OnBecameReportable(readHandler2);
 
-        readHandler1->ClearForceDirtyFlag();
+        // since the min interval on readHandler1 is 0, it should also be reportable now by sync mechanism
+        NL_TEST_ASSERT(aSuite, syncScheduler.IsReportableNow(readHandler1));
+        NL_TEST_ASSERT(aSuite, node1->GetSyncTimestamp() == node2->GetMinTimestamp());
+
+        // Confirm that the next report emission is scheduled on the min timestamp of readHandler2 as it is the highest reportable
+        NL_TEST_ASSERT(aSuite, syncScheduler.mTestNextReportTimestamp == node2->GetMinTimestamp());
+
         // Simulate a report emission for readHandler1
         readHandler1->mObserver->OnSubscriptionAction(readHandler1);
         NL_TEST_ASSERT(aSuite, !syncScheduler.IsReportableNow(readHandler1));
 
-        // ReadHandler 2 should still be reportable from the sync mechanism
+        // ReadHandler 2 should still be reportable since it hasn't emitted a report yet
         NL_TEST_ASSERT(aSuite, syncScheduler.IsReportableNow(readHandler2));
-        readHandler1->mObserver->OnSubscriptionAction(readHandler2);
+        readHandler2->ClearForceDirtyFlag();
```

Comment:
why do we need to manually clear the DirtyFlag?

Reply:
Because this would be done by the `SendReportData` after a report but we are not actually doing the engine run in this test so we clear manually.